### PR TITLE
sql/sem/tree: ensure the match type is pretty-printed as keyword

### PR DESCRIPTION
**Pull request by knz**
Release note: None

### CODE REVIEW

File: pkg/sql/sem/tree/testdata/pretty/create_table.align-deindent.golden [hunk 1]
```diff
@@ -68,7 +68,7 @@ CREATE TABLE product_information (
 		INT8
 		REFERENCES customers_2 (
 			id
-		)
+		) MATCH FULL
```
Comment (mjibson) [inline pkg/sql/sem/tree/testdata/pretty/create_table.align-deindent.golden]: How were these not printed before? I don't see how the change to pretty.go would add this text.

Comment (knz) [reply pkg/sql/sem/tree/testdata/pretty/create_table.align-deindent.golden]: I added it to the input SQL :)

**Review by madelynnblue — APPROVED** [review]

**Pull request merged**